pkg/urbit/worker/mars.c

```diff
@@ -1063,6 +1066,40 @@ _mars_boot_make(u3_boot_opts* inp_u,
       use = u3nc(u3nc(wir, cad), use);
     }
 
+    u3_noun mos = mor;
+    u3_noun pre = u3_nul;
+    u3_noun aft = u3_nul;
+    while ( u3_nul != mos ) {
+      u3_noun mot = u3h(mos);
+
+      switch ( u3h(mot) ) {
+        case c3__prop: {
+          u3_noun ter, met, ves;
+
+          if ( c3n == u3r_trel(u3t(mot), &met, &ter, &ves) ) {
+            u3m_p("invalid prop", u3t(mot));
+            break;
+          }
+
+          if ( c3y == u3r_sing_c("pre-userspace", ter) ) {
+            u3m_p("prop: pre-userspace", met);
+            pre = u3kb_weld(pre, ves);
+          }
+          else if ( c3y == u3r_sing_c("post-userspace", ter) ) {
+            u3m_p("prop: post-userspace", met);
+            aft = u3kb_weld(aft, ves);
+          }
+          else {
+            u3m_p("unrecognized prop tier", ter);
+          }
+        } break;
+
+        default: u3m_p("unrecognized boot sequence enhancement", u3h(mot));
+      }
+
+      mos = u3t(mos);
+    }
+
```

Comment:
this should all be in its own block (starting with `u3_noun mos = mor;`), and it should end with `u3z(mor)` (which is being leaked).